draft-ietf-dnsop-aname.md

```diff
@@ -271,8 +271,8 @@ The following steps MUST be performed for each address type:
 
  1. If one ore more address records are found, replace the owner of
     the target address records with the owner of the ANAME record.
-    Reduce the TTL to match the ANAME record if it is greater. Drop any
-    RRSIG records.
+    Set the TTL to the minimum of all TTLs of intermediate records.
```

Comment:
I realized after reading https://github.com/each/draft-aname/issues/30#issuecomment-488583324 that I misunderstood what was meant here by "intermediate records". The document should be updated to clarify (ideally both here and in Terminology), but also I disagree with exclusion of the ANAME itself. Just like `cname.example. 30 CNAME 60-second-ttl.example.` means that repeat queries for cname.example must not be fully answered from cache after 30 seconds, so should `aname.example. 30 ANAME 60-second-ttl.example.` mean that repeat queries for aname.example must not be fully answered from cache after 30 seconds, and therefore that the TTL of substitute records such as `aname.example. A 192.0.2.1` must not exceed the TTL of the ANAME responsible for them.

Reply:
First of all, it would be better if this discussion was held on the dnsop mailing list.

Second: Good point. To me intermediate records are records that are encountered when performing the ANAME resolution. The lookup encountered a CNAME or another ANAME. The TTLs of these intermediate records need to be taken into account when determining the final TTL of the address records to be substituted.

Your comment made me realize that there are two types of ANAME records: One that initializes a target lookup and others that are encountered during the target lookup. I think terminology for intermediate records and add text about whether the initial ANAME record should or should not be considered an intermediate record makes sense.

I still believe that the TTL on the initial ANAME should not be taken into the equation, but TTLs of intermediate ANAME records should. The reason for this is as I mentioned on the mailing list, the ANAME record may have different lifetime properties than its corresponding address records. Making them match means you will lose that.

Reply:
@gibson042 I pushed a commit to define intermediate records. Let me know if this looks good to you, or if you would like to see changes/additions, please send text :)

Reply:
Also, I haven't heard a counter argument against excluding the "initial ANAME" TTL, does that mean you agree to that behavior, or...?

Reply:
I do not agree with excluding the "initial ANAME" TTL, and my counterargument remains unchanged: 

> Just like `cname.example. 30 CNAME 60-second-ttl.example.` means that repeat queries for cname.example must not be fully answered from cache after 30 seconds [_but must instead be sent to name servers authoritative for cname.example in case the CNAME has been updated or removed_], so should `aname.example. 30 ANAME 60-second-ttl.example.` mean that repeat queries for aname.example must not be fully answered from cache after 30 seconds [_but must instead be sent to name servers authoritative for aname.example in case the ANAME has been updated or removed_], and therefore that the TTL of substitute records such as `aname.example. A 192.0.2.1` must not exceed the TTL of the ANAME responsible for them.

Reply:
I agree with @gibson042. if a query for example/A is answered via example/ANAME, which has a 30 second TTL, then there might be a different answer 31 seconds from now, so we should not answer from cache at that point.

Reply:
Right, I see now I was focused on not tying them together, because you want the ANAME TTL to be longer than the target address records TTL, but we are only updating (minimizing) the TTL of the address records. 

I am convinced now, and shall update the text.

Reply:
Actually, if the ANAME record had expired from cache, it cannot be answered from cache. Despite the address records are in cache for a long time, the resolver needs to look up the ANAME record first. So this can not happen anyway.

Reply:
FWIW I too think the initial ANAME TTL should be in the MIN(...) set.  I'd be quite surprised as an operator to discover that it wasn't.

Reply:
@matje This matters for originating clients. If an ANAME-ignorant caching resolver sends a request for aname.example and the receiving resolver has cached `aname.example. 30 ANAME 60-second-ttl.example.` (just refreshed) and `60-second-ttl.example. 42 AAAA 2001:db8::1` (aged down from 60 seconds), then it should synthesize `aname.example. 30 AAAA 2001:db8::1` rather than `aname.example. 42 AAAA 2001:db8::1` because the server for aname.example has only guaranteed that its address records should correspond with those of 60-second-ttl.example for 30 seconds (as indicated by the ANAME TTL), which means that synthesizing a response cacheable for longer than that (i.e., the 42-second remaining TTL of target records) is unjustified.